security/pkg/server/ca/authenticate/kubeauth/kube_jwt.go

```diff
@@ -112,18 +127,18 @@ func (a *KubeJWTAuthenticator) authenticate(targetJWT string, clusterID cluster.
 	// tolerate the unbound tokens.
 	if !util.IsK8SUnbound(targetJWT) || security.Require3PToken.Get() {
 		aud = security.TokenAudiences
-		if tokenAud, _ := util.ExtractJwtAud(targetJWT); len(tokenAud) == 1 && tokenAud[0] == DefaultKubernetesAudience {
+		if tokenAud, _ := util.ExtractJwtAud(targetJWT); isAllowedAudiences(tokenAud) {
```

Comment:
can we keep the len check? Then isAllowedAudiences can be simpler, like:

```go
if _, f := DefaultKubernetesAudience[tokenAud[0]; f {
 ....
}
```

And change DefaultKubernetesAudience to a map instead of a slice, then we can easily lookup.

Reply:
Thanks! Updated.